GitUI/CommandsDialogs/SettingsDialog/CheckSettingsLogic.cs

```diff
@@ -82,6 +82,15 @@ public bool SolveLinuxToolsDir(string? possibleNewPath = null)
 
                 if (CheckIfFileIsInPath("sh.exe") || CheckIfFileIsInPath("sh"))
                 {
+                    string configuredPath = AppSettings.GitBinDir;
+                    if (Directory.Exists(configuredPath))
+                    {
+                        if (File.Exists(configuredPath + "sh.exe") || File.Exists(configuredPath + "sh"))
+                        {
```

Comment:
Can join and reduce nesting

Reply:
In that case the checks on line 74, 93/102 should be changed too, they all follow the same pattern.
(also include line 83).
Adds noise to the change.

Reply:
I have inserted a refactoring commit.